From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Thu, 14 Apr 2022 10:58:25 +1000
Subject: [PATCH] Don't wander into non-ticking chunks

Original code by MultiPaper, licensed under GPL v3
You can find the original code on https://github.com/MultiPaper/MultiPaper

diff --git a/src/main/java/net/minecraft/world/entity/ai/goal/RandomStrollGoal.java b/src/main/java/net/minecraft/world/entity/ai/goal/RandomStrollGoal.java
index 676e1580849e8942eb41c7a3a80f464138b22764..9e527884c154c968ae0c93cb4282ea06f8ca3a24 100644
--- a/src/main/java/net/minecraft/world/entity/ai/goal/RandomStrollGoal.java
+++ b/src/main/java/net/minecraft/world/entity/ai/goal/RandomStrollGoal.java
@@ -2,6 +2,9 @@ package net.minecraft.world.entity.ai.goal;
 
 import java.util.EnumSet;
 import javax.annotation.Nullable;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.PathfinderMob;
 import net.minecraft.world.entity.ai.util.DefaultRandomPos;
 import net.minecraft.world.phys.Vec3;
@@ -49,7 +52,7 @@ public class RandomStrollGoal extends Goal {
             }
 
             Vec3 vec3 = this.getPosition();
-            if (vec3 == null) {
+            if (vec3 == null || !((ServerLevel) this.mob.level()).isPositionEntityTicking(BlockPos.containing(vec3))) { // MultiPaper - don't wander into non-ticking chunks
                 return false;
             } else {
                 this.wantedX = vec3.x;
diff --git a/src/main/java/net/minecraft/world/entity/animal/Squid.java b/src/main/java/net/minecraft/world/entity/animal/Squid.java
index c456e8959b4df057e04ca515e4f4b44f024a163f..5c2a9aad35097bcebcd6254af885e7f0f999bf62 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Squid.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Squid.java
@@ -192,6 +192,7 @@ public class Squid extends WaterAnimal {
             }
 
             if (!this.level().isClientSide) {
+                if ((tx != 0 || ty != 0 || tz != 0) && !((ServerLevel) this.level()).isPositionEntityTicking(BlockPos.containing(position().add(tx, ty, tz)))) tx = ty = tz = 0; // MultiPaper - don't allow squids to wander into non-ticking chunks
                 this.setDeltaMovement((double)(this.tx * this.speed), (double)(this.ty * this.speed), (double)(this.tz * this.speed));
             }
 
